hydradx/tests/test_global_state.py

```diff
@@ -12,3 +15,69 @@ def test_value_assets(market: dict, holdings: list):
     value = value_assets(market, assets)
     if value != sum([holdings[i] * market[asset_list[i]] for i in range(5)]):
         raise
+
+
+@given(
+    quantity=st.floats(min_value=100, max_value=1000000)
+)
+def test_external_market_sale(quantity: float):
+    initial_state = GlobalState(
+        pools={},
+        agents={
+            'agent': Agent(
+                holdings={
+                    'DAI': 1000000,
+                    'ETH': 1000
+                }
+            )
+        },
+        order_book={
+            'DAI': {1: 100000, 1.01: 100000, 1.02: 800000},
```

Comment:
And order book needs to have a "buy" and "sell" side (or bid/ask). See this notebook for structure of order book:
https://github.com/galacticcouncil/HydraDX-simulations/blob/omnipool-cex-arbitrage/hydradx/notebooks/Misc/arbitrage.ipynb

I also think we should keep the price/amount pairs in a list, I'm not sure we can guarantee that the prices are all unique. And we need to be able to keep the list sorted by price, probably sorted in descending order for the bids and ascending order for the asks.

For reference, here's the order book files we download from Kraken:
https://github.com/galacticcouncil/HydraDX-simulations/blob/omnipool-cex-arbitrage/hydradx/notebooks/Misc/data/dotusd_orderbook.json

Reply:
If the prices are not unique, wouldn't we just combine those that are the same? Or is there a reason not to do that?